crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1237,6 +1321,31 @@ where
             }
         }
     }
-
     None
 }
+
+async fn update_highest_sync_metrics<S>(
+    mut recv: oneshot::Receiver<()>,
+    store: S,
+    metrics: Metrics,
+) -> Result<()>
+where
+    S: WriteStore + Clone + Send + Sync,
+    <S as ReadStore>::Error: std::error::Error,
+{
+    let mut interval = tokio::time::interval(Duration::from_secs(5));
+    loop {
+        tokio::select! {
+             _now = interval.tick() => {
+                let highest_verified_checkpoint = store.get_highest_verified_checkpoint()
+                    .map_err(|e| anyhow!("Failed to update highest verified checkpoint with error: {:?}", e))?;
```

Comment:
we could just `.expect("store operation should not fail")` here since we do it elsewhere anyway